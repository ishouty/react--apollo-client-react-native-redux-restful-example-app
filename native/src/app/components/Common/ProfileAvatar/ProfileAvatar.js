import React, { Component } from 'react'
import { Image, StyleSheet, View } from 'react-native'
import PropTypes from 'prop-types'
import Layout from '../../../constants/Layout'

class ProfileAvatar extends Component {
  render() {
    const { sourceImage, size, callback } = this.props
    const style = size === undefined ? styles.medium : styles[size]

    return (
      <View className="inline">
        {callback ? (
          <div
            onClick={(e) => {
              callback(e)
            }}
          >
            <Image
              style={style}
              source={sourceImage}
              resizeMode="cover"
            />
          </div>
        ) : (
          <Image
            style={style}
            source={sourceImage}
            resizeMode="cover"
          />
        )}
      </View>
    )
  }
}

const styles = new StyleSheet.create({
  small: {
    width: 50,
    height: 50
  },
  medium: {
    width: Layout.width * 0.8,
    height: 50
  },
  large: {
    width: Layout.width * 0.8,
    height: 50
  }
})

ProfileAvatar.propTypes = {
  callback: PropTypes.func
}

export default ProfileAvatar
