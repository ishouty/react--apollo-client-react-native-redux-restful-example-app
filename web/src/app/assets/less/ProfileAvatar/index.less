.profile-avatar {
  border-radius: 50%;
  background-size: contain;
  background-repeat: no-repeat;
  display: inline-block;
  position: relative;
}

.profile-avatar-small {
  .profile-avatar;
  height: 20px;
  width: 20px;
  margin-right: 5px;
}

.profile-avatar-medium {
  .profile-avatar;
  height: 35px;
  width: 35px;
  margin-right: 10px;
}

.profile-avatar-large {
  .profile-avatar;
  max-height: 116px;
  width: 116px;
  height: 116px;
  margin: 0px;
}