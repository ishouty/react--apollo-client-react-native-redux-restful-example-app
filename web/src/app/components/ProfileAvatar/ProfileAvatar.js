import React, { Component } from 'react'

import '../../assets/less/ProfileAvatar/index.less'
import PropTypes from 'prop-types'

class ProfileAvatar extends Component {
  render() {
    const { profile, size, callback } = this.props

    if (!profile) return null

    const avatarClass = size === undefined ? 'medium' : size

    return (
      <div className="inline">
        {callback ? (
          <div
            className="profile-link"
            onClick={(e) => {
              callback(e)
            }}
          >
            <div
              className={'profile-avatar-' + avatarClass}
              style={{
                backgroundImage: `url(${profile.profileImage})`
              }}
            />
          </div>
        ) : (
          <div
            className={'profile-avatar-' + avatarClass}
            style={{
              backgroundImage: `url(${profile.profileImage})`
            }}
          />
        )}
      </div>
    )
  }
}

ProfileAvatar.propTypes = {
  profile: PropTypes.object.isRequired,
  nav: PropTypes.string,
  callback: PropTypes.func
}

export default ProfileAvatar
