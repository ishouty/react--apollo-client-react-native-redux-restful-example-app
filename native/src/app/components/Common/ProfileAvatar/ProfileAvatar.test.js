import 'react-native'
import React from 'react'
import ProfileAvatar from './ProfileAvatar'
import renderer from 'react-test-renderer'

it('renders component correctly', () => {
  const tree = renderer
    .create(
      <ProfileAvatar
        sourceImage={{ uri: 'mock.jpg' }}
        callback={() => {}}
      />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('profileAvatar will render the image larger', () => {
  const profileAvatarInstance = renderer.create(
    <ProfileAvatar
      sourceImage={{ uri: 'mock.jpg' }}
      callback={() => {}}
      size={'large'}
    />
  ).root

  profileAvatarInstance.findByType('Image').props.style ==
    { width: NaN, height: 50 }
})

it('profileAvatar will default to small image if no props has been entered', () => {
  const profileAvatarInstance = renderer.create(
    <ProfileAvatar
      sourceImage={{ uri: 'mock.jpg' }}
      callback={() => {}}
    />
  ).root

  profileAvatarInstance.findByType('Image').props.style ==
    { width: 50, height: 50 }
})
