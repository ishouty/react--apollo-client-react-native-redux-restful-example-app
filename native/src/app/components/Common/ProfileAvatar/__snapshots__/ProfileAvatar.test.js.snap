// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component correctly 1`] = `
<View
  className="inline"
>
  <div
    onClick={[Function]}
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": "mock.jpg",
        }
      }
      style={
        Object {
          "height": 50,
          "width": NaN,
        }
      }
    />
  </div>
</View>
`;
